core/data/clients/influx-client.go

```diff
@@ -0,0 +1,857 @@
+ * the License.
+ *
+ * @microservice: core-data-go library
+ * @author: Ryan Comer, Dell & Masataka Mizukoshi
+ * @version: 0.5.0
+ *******************************************************************************/
+
+package clients
+
+import (
+	"errors"
+	"fmt"
+	"encoding/json"
+	"github.com/edgexfoundry/edgex-go/core/domain/models"
+	"gopkg.in/mgo.v2/bson"
+	"github.com/influxdata/influxdb/client/v2"
+	"strconv"
+	"time"
+)
+
+var currentInfluxClient *InfluxClient // Singleton used so that InfluxEvent can use it to de-reference readings
+
+/*
+Core data client
+Has functions for interacting with the core data influxdb
+*/
+
+
+type InfluxClient struct {
+	Client   client.Client  // Influxdb client
+	Database string // Influxdb database name
+}
+
+// Return a pointer to the InfluxClient
+func newInfluxClient(config DBConfiguration) (*InfluxClient, error) {
+	// Create the dial info for the Influx session
+	connectionString := "http://" + config.Host + ":" + strconv.Itoa(config.Port)
+	influxdbHTTPInfo := client.HTTPConfig{
+		Addr:     connectionString,
+		Timeout:  time.Duration(config.Timeout) * time.Millisecond,
+		Username: config.Username,
+		Password: config.Password,
+	}
+	c, err := client.NewHTTPClient(influxdbHTTPInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	influxClient := &InfluxClient{Client: c, Database: config.DatabaseName}
+	currentInfluxClient = influxClient // Set the singleton
+	return influxClient, nil
+}
+
+// Get the current Influxdb Client
+func getCurrentInfluxClient() (*InfluxClient, error) {
+	if currentInfluxClient == nil {
+		return nil, errors.New("No current influx client, please create a new client before requesting it")
+	}
+
+	return currentInfluxClient, nil
+}
+
+// ******************************* EVENTS **********************************
+
+// Return all the events
+// UnexpectedError - failed to retrieve events from the database
+// Sort the events in descending order by ID
+func (ic *InfluxClient) Events() ([]models.Event, error) {
+	return ic.getEvents("")
+}
+
+// Add a new event
+// UnexpectedError - failed to add to database
+// NoValueDescriptor - no existing value descriptor for a reading in the event
+func (ic *InfluxClient) AddEvent(e *models.Event) (bson.ObjectId, error) {
+	e.Created = time.Now().UnixNano() / int64(time.Millisecond)
+	e.ID = bson.NewObjectId()
+
+	// Add the event
+	err := ic.addEventToDB(ic.Database, EVENTS_COLLECTION, e)
+	if err != nil {
+		return e.ID, err
+	}
+
+	return e.ID, err
+}
+
+// Update an event - do NOT update readings
+// UnexpectedError - problem updating in database
+// NotFound - no event with the ID was found
+func (ic *InfluxClient) UpdateEvent(e models.Event) error {
+	e.Modified = time.Now().UnixNano() / int64(time.Millisecond)
+
+	// Delete event
+	err := ic.deleteById(EVENTS_COLLECTION, e.ID.Hex())
+	if err != nil {
+		return ErrNotFound
+	}
+
+	// Add the event
+	err = ic.addEventToDB(ic.Database, EVENTS_COLLECTION, &e)
```

Comment:
You can just do `return ic.addEventToDB(ic.Database, EVENTS_COLLECTION, &e)`. 

This pattern is also found in some other functions.